src/nomad_material_processing/__init__.py

```diff
@@ -49,287 +52,364 @@
+        description="Section containing the crystal properties of the substrate.",
     )
     dopants = SubSection(
         section_def=Dopant,
         repeats=True,
-        description='''
+        description="""
         Repeating section containing information on any dopants in the substrate.
-        ''',
+        """,
     )
 
 
 class ThinFilm(CompositeSystem):
-    '''
+    """
     A thin film of material which exists as part of a stack.
-    '''
+    """
+
     m_def = Section()
 
     geometry = SubSection(
         section_def=Geometry,
-        description='Section containing the geometry of the thin film.',
+        description="Section containing the geometry of the thin film.",
+    )
+
+
+class ThinFilmReference(CompositeSystemReference):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section()
+    lab_id = Quantity(
+        type=str,
+        a_eln=ELNAnnotation(
+            component="StringEditQuantity",
+        ),
+    )
+    reference = Quantity(
+        type=ThinFilm,
+        a_eln=ELNAnnotation(
+            component="ReferenceEditQuantity",
+        ),
+    )
+
+
+class SubstrateReference(CompositeSystemReference):
+    """
+    A section for describing a system component and its role in a composite system.
+    """
+
+    lab_id = Quantity(
+        type=str,
+        a_eln=ELNAnnotation(
+            component="StringEditQuantity",
+        ),
+    )
+    reference = Quantity(
+        type=SubstrateSlab,
+        a_eln=ELNAnnotation(
+            component="ReferenceEditQuantity",
+        ),
     )
 
 
 class ThinFilmStack(CompositeSystem):
-    '''
+    """
     A stack of `ThinFilm`(s). Typically deposited on a `Substrate`.
-    '''
+    """
+
     m_def = Section()
-    substrate = Quantity(
-        type=Substrate,
-        description='''
+    # layers = Quantity(
+    #     type=ThinFilm,
+    #     description='''
+    #     An ordered list (starting at the substrate) of the thin films making up the
+    #     thin film stacks.
+    #     ''',
+    #     a_eln=ELNAnnotation(
+    #         component=ELNComponentEnum.ReferenceEditQuantity,
+    #     ),
+    #     shape=["*"],
+    # )
+    layers = SubSection(
+        description="""
+        An ordered list (starting at the substrate) of the thin films making up the
+        thin film stacks.
+        """,
+        section_def=ThinFilmReference,
+        repeats=True,
+    )
+    substrate = SubSection(
+        description="""
         The substrate which the thin film layers of the thin film stack are deposited
         on.
-        ''',
+        """,
+        section_def=SubstrateReference,
+        repeats=True,
+    )
+    ####################################################################################
+
+
+########  !!! Here I see layers and substarte. What about components subsection present in CompositeSystem class?  ####TODO
```

Comment:
Yes, we should add a normalize function that populates the components subsection and possibly hide that section.

Reply:
I will draft one in this MR